### PR TITLE
feat(ST-94): add assignmets display section and integrate api

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
## Description of Change

- Added assignment card component
- Added get assignments rtk query inside assignments.api.ts
- Integrated it in resources container
- Rendered assignments

## Associated Ticket Link(s)

- https://trello.com/c/Gt1yGn83

## Related Pull Request(s)

- N/A

## Screenshots / Screen Recordings
Web view:
Screen Shot 2024-09-23 at 11 40 46 PM

Tab view:
Screen Shot 2024-09-23 at 11 41 41 PM

Mobile view:
Screen Shot 2024-09-23 at 11 41 27 PM

### CODE REVIEW

**Comment by ragib-hossain-sazim** [comment]
[Nitpick] Typo on commit message. `Ingrate` => `Integrate`

**Review by ragib-hossain-sazim — APPROVED** [review]

File: modules/Classroom/containers/ClassroomResourcesContainer/ClassroomResourcesContainer.tsx [hunk 1]
```diff
@@ -3,20 +3,27 @@ import { useDisclosure } from "@mantine/hooks";
...
   const [isMaterialsOpened, { toggle: materialsToggle }] = useDisclosure(false);
+  const [isAssignmentsOpened, { toggle: assignmentsToggle }] = useDisclosure(false);
+
   const { data: fetchedMaterials, isLoading: isMaterialsLoading } = useGetResourceMaterialsQuery(
     classroomId,
     { skip: !classroomId },
   );
 
+  const { data: fetchedAssignments, isLoading: isAssignmentsLoading } =
+    useGetAssignmentsQuery(classroomId);
```
Comment (ishmam-iqbal-sazim) [inline modules/Classroom/containers/ClassroomResourcesContainer/ClassroomResourcesContainer.tsx:25]: Query should be skipped if classroomId is falsy

File: modules/Classroom/containers/ClassroomResourcesContainer/ClassroomResourcesContainer.tsx [hunk 2]
```diff
@@ -50,6 +57,31 @@ const ClassroomResourcesContainer = ({ classroomId }: TClassroomResourcesContain
...
+            <FaAngleRight className={classes.icon} onClick={assignmentsToggle} />
+          )}
+          <Title order={4}>Assignments</Title>
+        </Flex>
+        {isAssignmentsLoading ? (
+          <LoadingComponent visible />
+        ) : (
+          <Box className={classes.cardsContainer}>
+            {isAssignmentsOpened
+              ? fetchedAssignments?.map((assignment) => (
```
Comment (ishmam-iqbal-sazim) [inline modules/Classroom/containers/ClassroomResourcesContainer/ClassroomResourcesContainer.tsx:74]: What if fetchedAssginments length is zero?

File: modules/Classroom/containers/ClassroomResourcesContainer/ClassroomResourcesContainer.tsx [hunk 3]
```diff
@@ -50,6 +57,31 @@ const ClassroomResourcesContainer = ({ classroomId }: TClassroomResourcesContain
...
+                  <AssignmentCard
+                    key={assignment.id}
+                    assignment={assignment}
+                    classroomId={classroomId}
+                  />
+                ))
+              : null}
```
Comment (ishmam-iqbal-sazim) [inline modules/Classroom/containers/ClassroomResourcesContainer/ClassroomResourcesContainer.tsx:81]: Let's try to avoid nested ternaries

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments

**Pull request merged**